2020 Florida Congressional Districts

geogar3 opened the pull request:
## Redistricting requirements
In Florida, according to [the state constitution Art. III §§ 20](http://www.leg.state.fl.us/statutes/index.cfm?submenu=3#A3S20), districts must:
1. not be drawn with the intent to favor or disfavor a political party or an incumbent
2. not be drawn with the intent or result of denying or abridging the electoral opportunities of racial or language minorities
(The following are required in so much as they do not impose on the above requirements)
3. be as nearly equal in population as is practicable
4. be compact
5. utilize existing political and geographical boundaries
6. preserve county and municipality boundaries as much as possible


### Interpretation of requirements
We enforce a maximum population deviation of 2%.

## Data Sources
Data for Florida comes from the ALARM Project's [2020 Redistricting Data Files](https://alarm-redist.github.io/posts/2021-08-10-census-2020/).
Data for Florida's 2020 congressional district map comes from the [Dave's Redistricting](https://davesredistricting.org/maps#home)

## Pre-processing Notes
We estimate CVAP populations with the [cvap](https://github.com/christopherkenny/cvap) R package. We also pre-process the map to split it into clusters for simulation, which has a slight effect on the types of redistrict plans that will be sampled.

## Simulation Notes
We sample 35,000 districting plans for Florida across two independent runs of the SMC algorithm, and then thin the sample down to 5,000 plans. Due to the size, shape, and complexity of Florida, we split the simulations into multiple steps.

1. Regional clustering
First, we cluster Florida counties into 3 regions--Southern Florida, Northern Florida, and Central Florida--with the following county assignments:

Southern Florida: Broward, Charlotte, Collier, DeSoto, Glades, Hardee, Hendry, Highlands, Lee, Manatee, Martin, Miami-Dade, Monroe, Okeechobee, Palm Beach, Sarasota, and St. Lucie

Northern Florida: Alachua, Baker, Bay, Bradford, Calhoun, Clay, Columbia, Dixie, Duval, Escambia, Franklin, Gadsden, Gilchrist, Gulf, Hamilton, Holmes, Jackson, Jefferson, Lafayette, Leon, Levy, Liberty, Madison, Marion, Nassau, Okaloosa, Putnam, Santa Rosa, St. Johns, Suwannee, Taylor, Union, Wakulla, Walton, and Washington

Central Florida: Brevard, Citrus, Flagler, Hernando, Hillsborough, Indian River, Lake, Lake, Orange, Osceola, Pasco, Pinellas, Polk, Seminole, Sumter, and Volusia

County assignments were based on the collections of counties that define Metropolitan and Combined Statistical Areas and on past and current Congressional district maps.

2. Simulating Northern and Southern Florida
We run simulations first in Northern and Southern Florida. These simulations run the SMC algorithm within each cluster with a 1.75% population tolerance. Because each cluster will have leftover population, we apply an additional constraint that encourages unassigned areas to be set on each cluster's border with the Central Florida cluster, thereby avoiding district discontiguities.

In both the Northern Florida cluster and the Southern Florida cluster, we apply Gibbs constraints to encourage the formation of Black and Hispanic opportunity districts. To balance county and municipality splits, we create pseudocounties for use in the county constraint, which leads to fewer municipality splits than using only a county constraint.

3. Simulating Central Florida
The partial map simulations from the Southern and Northern Florida clusters are then combined, with unassigned areas being absorbed into the Central Florida cluster. We then run simulations in Central Florida, applying Gibbs hinge constraints to encourage the formation of minority opportunity districts. To limit county and municipality splits, we create pseudocounties for use in the county constraint.


## Validation
![validation_20220628_1406](https://user-images.githubusercontent.com/90931177/176289006-2eea961e-d920-4396-9fc8-88c6211eba61.png)
![vap_histograms](https://user-images.githubusercontent.com/90931177/176289327-643833bd-6568-4772-8906-309876a9c8fc.png)

```
SMC: 5,000 sampled plans of 28 districts on 7,211 units
`adapt_k_thresh`=0.985 • `seq_alpha`=0.65
`est_label_mult`=1 • `pop_temper`=0.05

Plan diversity 80% range: 0.37 to 0.76

R-hat values for summary statistics:
     total_vap     total_cvap       plan_dev      comp_edge    comp_polsby       pop_hisp      pop_white      pop_black       pop_aian 
     1.0030774      1.0094615      1.0208456      1.0126730      1.0006476      1.0325589      1.0335017      1.0005033      1.0202896 
     pop_asian       pop_nhpi      pop_other        pop_two       vap_hisp      vap_white      vap_black       vap_aian      vap_asian 
     1.0019691      1.0144814      1.0009915      1.0104842      1.0299505      1.0331114      1.0004405      1.0200506      1.0023774 
      vap_nhpi      vap_other        vap_two     cvap_white     cvap_black      cvap_hisp     cvap_asian      cvap_aian      cvap_nhpi 
     1.0265273      1.0038276      1.0109165      1.0335174      1.0043058      1.0318200      1.0011201      1.0066325      1.0107006 
      cvap_two     cvap_other pre_16_rep_tru pre_16_dem_cli uss_16_rep_rub uss_16_dem_mur uss_18_rep_sco uss_18_dem_nel gov_18_rep_des 
     1.0015892      1.0010557      1.0164486      1.0010196      1.0029982      1.0024271      1.0035176      1.0045065      1.0134324 
gov_18_dem_gil atg_18_rep_moo atg_18_dem_sha pre_20_rep_tru pre_20_dem_bid         arv_16         adv_16         arv_18         adv_18 
     1.0037674      1.0131786      1.0029661      0.9999539      1.0044187      1.0124261      1.0016869      1.0135115      1.0037281 
        arv_20         adv_20  county_splits    muni_splits            ndv            nrv        ndshare          e_dvs         pr_dem 
     0.9999539      1.0044187      1.0268956      1.0222689      1.0028136      1.0039284      1.0012198      1.0012408      1.0042675 
         e_dem          pbias           egap 
     1.0009490      1.0074459      1.0026318 

Sampling diagnostics for SMC run 1 of 2 (35,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    19,865 (56.8%)     26.9%        0.72 21,996 ( 99%)     17 
Split 2    17,521 (50.1%)     42.3%        0.87 19,637 ( 89%)      9 
Split 3    16,951 (48.4%)     45.7%        0.92 19,076 ( 86%)      7 
Split 4    15,263 (43.6%)     51.1%        0.93 18,750 ( 85%)      5 
Split 5    15,670 (44.8%)     36.2%        0.91 18,233 ( 82%)      7 
Split 6    15,003 (42.9%)     36.2%        0.92 18,065 ( 82%)      6 
Split 7    16,607 (47.4%)     41.7%        0.90 17,799 ( 80%)      4 
Split 8    16,020 (45.8%)     31.0%        0.91 17,659 ( 80%)      5 
Split 9    16,923 (48.4%)     30.6%        0.91 17,219 ( 78%)      4 
Split 10   17,517 (50.0%)     19.8%        0.86 16,934 ( 77%)      5 
Split 11   15,178 (43.4%)      8.4%        0.82 15,381 ( 70%)      4 
Resample    3,564 (10.2%)       NA%        1.44 13,114 ( 59%)     NA 

Sampling diagnostics for SMC run 2 of 2 (35,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    19,540 (55.8%)     27.2%        0.72 22,056 (100%)     17 
Split 2    17,505 (50.0%)     35.5%        0.86 19,686 ( 89%)     11 
Split 3    16,338 (46.7%)     45.3%        0.90 18,977 ( 86%)      7 
Split 4    14,630 (41.8%)     51.4%        0.93 18,675 ( 84%)      5 
Split 5    14,835 (42.4%)     40.6%        0.92 18,180 ( 82%)      6 
Split 6    15,072 (43.1%)     35.7%        0.91 17,928 ( 81%)      6 
Split 7    16,046 (45.8%)     18.3%        0.90 17,554 ( 79%)     11 
Split 8    17,320 (49.5%)     23.3%        0.87 17,733 ( 80%)      7 
Split 9    14,081 (40.2%)     25.5%        0.88 17,691 ( 80%)      5 
Split 10   17,166 (49.0%)     12.0%        0.88 16,819 ( 76%)      9 
Split 11   15,905 (45.4%)      7.9%        0.89 15,248 ( 69%)      5 
Resample     2,889 (8.3%)       NA%        1.32 13,610 ( 62%)     NA 
```

## Checklist
- [x] I have followed the [instructions](https://github.com/alarm-redist/fifty-states/blob/main/CONTRIBUTING.md)
- [x] I have updated the [tracker](https://docs.google.com/spreadsheets/d/1k_tYLoE49W_DCK1tcWbouoYZFI9WD76oayEt5TOmJg4/edit#gid=453387933)
- [x] All `TODO` lines from the template code have been removed
- [x] I have merged in the master branch and then recalculated summary statistics
- [x] I have run `enforce_style()` to format my code
- [x] The documentation copied above is up-to-date 
- [x] There are no data files in this pull request
- [x] None of the file output paths (for the `redist_map` and `redist_plans` objects, and summary statistics) have been edited

**delete this line and all the tags except the reviewers you need**
@christopherkenny
@tylersimko
@CoryMcCartan

geogar3 commented:
Validation plots including Desantis' plan:

tylersimko commented:
Thanks @geogar3! Great work here, I know this is a crazy one. Two questions about the BVAP districts:

1.Looks like you're stuck between 30-40% BVAP based on the histograms. Just curious if you know where the bulk of the distributions fall within that range for the two districts (e.g. are we talking 30.7% or like 37%). 
2. Are they roughly in the right geographic area (are you trying to match the two southern districts?)

CoryMcCartan commented:
Thanks @geogar3 this was a lot of work.
Do you have the `summary()` info for N and S Florida?

christopherkenny commented on analyses/FL_cd_2020/03_sim_FL_cd_2020.R:
```diff
@@ -0,0 +1,173 @@
+    add_constr_grp_hinge(10, vap_black, vap, 0.4) %>%
+    add_constr_grp_hinge(-12, vap_black, vap, 0.1) %>%
+    # reward districts with hispanic vap % below 30% and above 70%, enforcing barrier in between
+    add_constr_grp_hinge(5, vap_hisp, vap, 0.7) %>%
```
can we use `inv_hinge` here? If the goal is to keep it below, this is nudging it too high, which may contribute to the diversity issues here.

geogar3 replied:
> `inv_hinge` throughout or on any of the constraints specifically?

christopherkenny commented on analyses/FL_cd_2020/03_sim_FL_cd_2020.R:
```diff
@@ -0,0 +1,173 @@
+
+########################################################################
+
+constraints <- redist_constr(map_south) %>%
```
the weights here may need to come down in general to avoid penalizing outside of the target zone too much.

christopherkenny commented on analyses/FL_cd_2020/03_sim_FL_cd_2020.R:
```diff
@@ -0,0 +1,173 @@
+
+n_steps <- (sum(map_south$pop)/attr(map, "pop_bounds")[2]) %>% floor()
+
+plans_south <- redist_smc(map_south,
```
I think much of the difficulty is coming out of here. Based on the example maps and the plans_diversity, there are some districts in the south that are getting repeated a lot, while the north is fine. What does the summary(plans_south) look like?

geogar3 replied:
I agree that plans_south is the issue. I've had a hell of a time trying to increase diversity without losing the minority opportunity districts. See the summary below.

christopherkenny commented on analyses/FL_cd_2020/03_sim_FL_cd_2020.R:
```diff
@@ -0,0 +1,173 @@
+
+set.seed(2020)
+
+cluster_pop_tol <- 0.0175
```
I'm a little worried about how big this is. Can we reduce this at all?

christopherkenny commented on analyses/FL_cd_2020/03_sim_FL_cd_2020.R:
```diff
@@ -0,0 +1,173 @@
+    counties = pseudo_county,
+    constraints = constraints,
+    init_particles = prep_mat,
+    pop_temper = 0.05, seq_alpha = 0.65)  %>%
```
pop_temper might be too high, which could be hurting this if it's making the first few splits struggle

geogar3 commented:
@tylersimko 
_1.Looks like you're stuck between 30-40% BVAP based on the histograms. Just curious if you know where the bulk of the distributions fall within that range for the two districts (e.g. are we talking 30.7% or like 37%). 
2. Are they roughly in the right geographic area (are you trying to match the two southern districts?). So, just verifying these two near BVAP districts appear in the right submap._
> The bulk of the distribution is falling <35% but do appear to be in roughly the right geographic area (labeled districts in the plot are matched with cd_2020 districts)
![image](https://user-images.githubusercontent.com/90931177/176750411-5ad6bc93-d715-4ec6-8de8-0a66c64d6ed5.png)

geogar3 commented:
@CoryMcCartan 
_Do you have the summary() info for N and S Florida?_
For some reason, it won't show the R-hats for the summary statistics I included (bvap, hvap, dem16, dem18, dem20--same as what Tyler included in his subplans). Can't figure out why. Here are the other diagnostics:
```
> summary(plans_south)
SMC: 35,000 sampled plans of 12 districts on 3,243 units
`adapt_k_thresh`=0.985 • `seq_alpha`=0.65
`est_label_mult`=1 • `pop_temper`=0.06

Plan diversity 80% range: 0.069 to 0.801

Sampling diagnostics for SMC run 1 of 1 (35,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    16,695 (47.7%)     22.1%        0.65 22,089 (100%)     22 
Split 2    16,329 (46.7%)     30.2%        0.69 16,950 ( 77%)     14 
Split 3    14,699 (42.0%)     30.7%        0.71 16,842 ( 76%)     12 
Split 4    13,051 (37.3%)     22.2%        0.76 16,350 ( 74%)     15 
Split 5    11,540 (33.0%)     29.8%        0.81 15,665 ( 71%)     10 
Split 6    10,848 (31.0%)     29.9%        0.87 15,139 ( 68%)      9 
Split 7    10,138 (29.0%)     41.9%        0.87 15,146 ( 68%)      5 
Split 8     9,112 (26.0%)     21.8%        0.89 14,616 ( 66%)     10 
Split 9     6,729 (19.2%)     28.4%        0.95 14,230 ( 64%)      6 
Split 10   10,310 (29.5%)     30.8%        1.00 13,336 ( 60%)      4 
Split 11   11,598 (33.1%)     32.1%        0.84 11,358 ( 51%)      5 
Resample    4,700 (13.4%)       NA%        8.22 11,628 ( 53%)     NA 
```

```
> summary(plans_north)
SMC: 35,000 sampled plans of 6 districts on 1,608 units
`adapt_k_thresh`=0.985 • `seq_alpha`=0.5
`est_label_mult`=1 • `pop_temper`=0

Plan diversity 80% range: 0.36 to 0.55

Sampling diagnostics for SMC run 1 of 1 (35,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    29,781 (85.1%)     21.1%        0.81 22,053 (100%)     14 
Split 2    27,264 (77.9%)     44.9%        0.71 20,553 ( 93%)      9 
Split 3    23,849 (68.1%)     47.5%        0.99 18,764 ( 85%)      6 
Split 4    23,125 (66.1%)     37.7%        0.93 17,188 ( 78%)      8 
Split 5    14,287 (40.8%)     58.1%        0.68 18,805 ( 85%)      5 
Resample    6,301 (18.0%)       NA%       10.21 12,107 ( 55%)     NA 
```

geogar3 commented:
## Redistricting requirements
In Florida, according to [the state constitution Art. III §§ 20](http://www.leg.state.fl.us/statutes/index.cfm?submenu=3#A3S20), districts must:
1. not be drawn with the intent to favor or disfavor a political party or an incumbent
2. not be drawn with the intent or result of denying or abridging the electoral opportunities of racial or language minorities
(The following are required in so much as they do not impose on the above requirements)
3. be as nearly equal in population as is practicable
4. be compact
5. utilize existing political and geographical boundaries
6. preserve county and municipality boundaries as much as possible


### Interpretation of requirements
We enforce a maximum population deviation of 2%.

## Data Sources
Data for Florida comes from the ALARM Project's [2020 Redistricting Data Files](https://alarm-redist.github.io/posts/2021-08-10-census-2020/).
Data for Florida's 2020 congressional district map comes from the [Dave's Redistricting](https://davesredistricting.org/maps#home)

## Pre-processing Notes
We estimate CVAP populations with the [cvap](https://github.com/christopherkenny/cvap) R package. We also pre-process the map to split it into clusters for simulation, which has a slight effect on the types of redistrict plans that will be sampled.

## Simulation Notes
We sample 160,000 districting plans for Florida across two independent runs of the SMC algorithm, and then thin the sample down to 5,000 plans. Due to the size, shape, and complexity of Florida, we split the simulations into multiple steps.

1. Regional clustering
First, we cluster Florida counties into 3 regions--Southern Florida, Northern Florida, and Central Florida--with the following county assignments:

Southern Florida: Broward, Charlotte, Collier, DeSoto, Glades, Hardee, Hendry, Highlands, Lee, Manatee, Martin, Miami-Dade, Monroe, Okeechobee, Palm Beach, Sarasota, and St. Lucie

Northern Florida: Alachua, Baker, Bay, Bradford, Calhoun, Clay, Columbia, Dixie, Duval, Escambia, Franklin, Gadsden, Gilchrist, Gulf, Hamilton, Holmes, Jackson, Jefferson, Lafayette, Leon, Levy, Liberty, Madison, Marion, Nassau, Okaloosa, Putnam, Santa Rosa, St. Johns, Suwannee, Taylor, Union, Wakulla, Walton, and Washington

Central Florida: Brevard, Citrus, Flagler, Hernando, Hillsborough, Indian River, Lake, Lake, Orange, Osceola, Pasco, Pinellas, Polk, Seminole, Sumter, and Volusia

County assignments were based on the collections of counties that define Metropolitan and Combined Statistical Areas and on past and current Congressional district maps.

2. Simulating Northern and Southern Florida
We run simulations first in Northern and Southern Florida. These simulations run the SMC algorithm within each cluster with a 1.75% population tolerance. Because each cluster will have leftover population, we apply an additional constraint that encourages unassigned areas to be set on each cluster's border with the Central Florida cluster, thereby avoiding district discontiguities.

In both the Northern Florida cluster and the Southern Florida cluster, we apply Gibbs constraints to encourage the formation of Black and Hispanic opportunity districts. To balance county and municipality splits, we create pseudocounties for use in the county constraint, which leads to fewer municipality splits than using only a county constraint.

3. Simulating Central Florida
The partial map simulations from the Southern and Northern Florida clusters are then combined, with unassigned areas being absorbed into the Central Florida cluster. We then run simulations in Central Florida, applying Gibbs hinge constraints to encourage the formation of minority opportunity districts. To limit county and municipality splits, we create pseudocounties for use in the county constraint.


## Validation
![validation_20220711_1300](https://user-images.githubusercontent.com/90931177/179283786-39d9053a-054b-439a-ba74-d836de6c5ff1.png)

```
> summary(plans)
SMC: 5,000 sampled plans of 28 districts on 7,211 units
`adapt_k_thresh`=0.985 • `seq_alpha`=0.65
`est_label_mult`=1 • `pop_temper`=0.05

Plan diversity 80% range: 0.56 to 0.77

R-hat values for summary statistics:
     total_vap     total_cvap       plan_dev      comp_edge    comp_polsby       pop_hisp      pop_white      pop_black       pop_aian 
      1.035412       1.013526       1.007595       1.002202       1.007033       1.005974       1.011126       1.023203       1.006439 
     pop_asian       pop_nhpi      pop_other        pop_two       vap_hisp      vap_white      vap_black       vap_aian      vap_asian 
      1.005153       1.012094       1.002750       1.006343       1.003374       1.015567       1.033380       1.010479       1.005329 
      vap_nhpi      vap_other        vap_two     cvap_white     cvap_black      cvap_hisp     cvap_asian      cvap_aian      cvap_nhpi 
      1.017295       0.999864       1.002161       1.015432       1.017921       1.004134       1.000512       1.003259       1.027074 
      cvap_two     cvap_other pre_16_rep_tru pre_16_dem_cli uss_16_rep_rub uss_16_dem_mur uss_18_rep_sco uss_18_dem_nel gov_18_rep_des 
      1.009758       1.026852       1.013806       1.005240       1.008317       1.004335       1.000585       1.008602       1.000764 
gov_18_dem_gil atg_18_rep_moo atg_18_dem_sha pre_20_rep_tru pre_20_dem_bid         arv_16         adv_16         arv_18         adv_18 
      1.008496       1.000385       1.008191       1.000102       1.010573       1.011360       1.004528       1.000861       1.008468 
        arv_20         adv_20  county_splits    muni_splits            ndv            nrv        ndshare          e_dvs          e_dem 
      1.000102       1.010573       1.019671       1.018568       1.008194       1.002099       1.025007       1.026144       1.022008 
         pbias           egap    pop_overlap 
      1.030323       1.014046       1.015560 

Sampling diagnostics for SMC run 1 of 2 (80,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    58,835 (73.5%)     18.9%        0.72 49,679 ( 98%)      8 
Split 2    51,453 (64.3%)     18.9%        0.81 46,057 ( 91%)      6 
Split 3    45,633 (57.0%)     15.9%        0.88 44,292 ( 88%)      6 
Split 4    42,793 (53.5%)     16.5%        0.91 42,999 ( 85%)      5 
Split 5    42,217 (52.8%)     17.6%        0.89 42,436 ( 84%)      4 
Split 6    41,096 (51.4%)     19.9%        0.88 41,914 ( 83%)      3 
Split 7    40,528 (50.7%)     13.2%        0.87 41,027 ( 81%)      4 
Split 8    41,722 (52.2%)     11.1%        0.87 40,582 ( 80%)      4 
Split 9    42,493 (53.1%)     11.8%        0.88 39,976 ( 79%)      3 
Split 10   39,044 (48.8%)      9.4%        0.89 38,437 ( 76%)      3 
Split 11   34,478 (43.1%)      4.9%        0.87 34,318 ( 68%)      2 
Resample   10,127 (12.7%)       NA%        1.40 29,836 ( 59%)     NA 

Sampling diagnostics for SMC run 2 of 2 (80,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    58,299 (72.9%)     18.9%        0.71 49,463 ( 98%)      8 
Split 2    50,987 (63.7%)     22.5%        0.81 45,941 ( 91%)      5 
Split 3    46,471 (58.1%)     23.3%        0.88 44,474 ( 88%)      4 
Split 4    41,919 (52.4%)     20.2%        0.90 43,152 ( 85%)      4 
Split 5    41,445 (51.8%)     17.7%        0.90 41,958 ( 83%)      4 
Split 6    40,983 (51.2%)     19.9%        0.89 41,524 ( 82%)      3 
Split 7    40,944 (51.2%)     23.8%        0.87 41,282 ( 82%)      2 
Split 8    42,424 (53.0%)     19.9%        0.86 41,225 ( 82%)      2 
Split 9    43,535 (54.4%)     16.4%        0.86 40,306 ( 80%)      2 
Split 10   44,336 (55.4%)     13.4%        0.86 38,736 ( 77%)      2 
Split 11   35,867 (44.8%)      2.6%        0.82 35,294 ( 70%)      4 
Resample     5,884 (7.4%)       NA%        1.41 29,144 ( 58%)     NA 
```

```
> summary(plans_south)
SMC: 80,000 sampled plans of 12 districts on 3,243 units
`adapt_k_thresh`=0.985 • `seq_alpha`=0.65
`est_label_mult`=1 • `pop_temper`=0.06

Plan diversity 80% range: 0.66 to 0.85

R-hat values for summary statistics:
    hvap     bvap    dem16    dem18    dem20 
1.002034 1.002779 1.027016 1.034260 1.040566 

Sampling diagnostics for SMC run 1 of 2 (40,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    30,809 (77.0%)     11.7%        0.63 25,362 (100%)     10 
Split 2    28,898 (72.2%)     17.1%        0.63 23,460 ( 93%)      6 
Split 3    28,413 (71.0%)     21.9%        0.64 22,956 ( 91%)      4 
Split 4    26,992 (67.5%)     24.9%        0.66 22,587 ( 89%)      3 
Split 5    25,365 (63.4%)     22.1%        0.70 22,141 ( 88%)      3 
Split 6    24,668 (61.7%)     26.3%        0.72 21,664 ( 86%)      2 
Split 7    23,396 (58.5%)     23.2%        0.74 21,120 ( 84%)      2 
Split 8    20,540 (51.4%)     14.8%        0.76 20,626 ( 82%)      3 
Split 9    19,065 (47.7%)     16.5%        0.79 19,574 ( 77%)      2 
Split 10   16,287 (40.7%)     13.9%        0.83 18,248 ( 72%)      2 
Split 11    9,599 (24.0%)     11.3%        0.88 14,373 ( 57%)      2 
Resample     3,619 (9.0%)       NA%        8.14 10,840 ( 43%)     NA 

Sampling diagnostics for SMC run 2 of 2 (40,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    30,789 (77.0%)     16.6%        0.63 25,278 (100%)      7 
Split 2    28,874 (72.2%)     20.2%        0.63 23,375 ( 92%)      5 
Split 3    28,365 (70.9%)     13.2%        0.64 22,887 ( 91%)      7 
Split 4    26,757 (66.9%)     16.1%        0.67 22,562 ( 89%)      5 
Split 5    25,552 (63.9%)     22.3%        0.71 22,047 ( 87%)      3 
Split 6    24,477 (61.2%)     19.8%        0.72 21,642 ( 86%)      3 
Split 7    22,593 (56.5%)     23.2%        0.75 21,001 ( 83%)      2 
Split 8    21,944 (54.9%)     19.6%        0.75 20,574 ( 81%)      2 
Split 9    20,291 (50.7%)     16.1%        0.78 19,730 ( 78%)      2 
Split 10   16,669 (41.7%)     13.2%        0.79 18,556 ( 73%)      2 
Split 11   10,290 (25.7%)     10.3%        0.87 15,004 ( 59%)      2 
Resample     3,349 (8.4%)       NA%        8.12 10,138 ( 40%)     NA 
```
```
> summary(plans_north)
SMC: 80,000 sampled plans of 6 districts on 1,608 units
`adapt_k_thresh`=0.985 • `seq_alpha`=0.5
`est_label_mult`=1 • `pop_temper`=0

Plan diversity 80% range: 0.36 to 0.55

R-hat values for summary statistics:
     hvap      bvap     dem16     dem18     dem20 
1.0006607 1.0004837 0.9999888 0.9999899 0.9999878 

Sampling diagnostics for SMC run 1 of 2 (40,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    35,137 (87.8%)     10.1%        0.78 25,308 (100%)      8 
Split 2    34,536 (86.3%)     23.6%        0.68 23,306 ( 92%)      5 
Split 3    30,451 (76.1%)     20.6%        0.88 21,829 ( 86%)      4 
Split 4    26,660 (66.7%)     18.4%        0.88 20,358 ( 81%)      4 
Split 5    16,357 (40.9%)     23.0%        0.65 21,270 ( 84%)      4 
Resample    8,015 (20.0%)       NA%       10.19 14,156 ( 56%)     NA 

Sampling diagnostics for SMC run 2 of 2 (40,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd   Max. unique Est. k 
Split 1    35,075 (87.7%)     11.6%        0.79 25,259 (100%)      7 
Split 2    34,457 (86.1%)     23.6%        0.69 23,162 ( 92%)      5 
Split 3    30,415 (76.0%)     16.9%        0.89 21,766 ( 86%)      5 
Split 4    26,768 (66.9%)     10.8%        0.88 20,153 ( 80%)      7 
Split 5    16,159 (40.4%)     22.9%        0.65 21,354 ( 84%)      4 
Resample    4,691 (11.7%)       NA%       10.21 14,018 ( 55%)     NA 
```

## Checklist
- [x] I have followed the [instructions](https://github.com/alarm-redist/fifty-states/blob/main/CONTRIBUTING.md)
- [x] I have updated the [tracker](https://docs.google.com/spreadsheets/d/1k_tYLoE49W_DCK1tcWbouoYZFI9WD76oayEt5TOmJg4/edit#gid=453387933)
- [x] All `TODO` lines from the template code have been removed
- [x] I have merged in the master branch and then recalculated summary statistics
- [x] I have run `enforce_style()` to format my code
- [x] The documentation copied above is up-to-date 
- [x] There are no data files in this pull request
- [x] None of the file output paths (for the `redist_map` and `redist_plans` objects, and summary statistics) have been edited

**delete this line and all the tags except the reviewers you need**
@christopherkenny
@tylersimko
@CoryMcCartan 

## Other diagnostic plots
![bvap_plot](https://user-images.githubusercontent.com/90931177/179284450-6cc20a22-9115-4e4c-91c8-e814ee2ad559.png)
![vap_histograms](https://user-images.githubusercontent.com/90931177/179284457-fb718dad-52ff-4017-9614-91df7106ddcf.png)
![performance_plots](https://user-images.githubusercontent.com/90931177/179284463-a22a3106-1a25-4eae-b39c-7d9b695f002c.png)

CoryMcCartan commented:
Ready to go for me!

christopherkenny commented:
Good to go by me. @tylersimko, on to you!
